hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -43,6 +44,40 @@
 from hpat.utils import to_array
 
 
+class TypingChecker:
+    msg_template = '{} The object must be {}. Given{}: {}'
+    integer_types = (types.Omitted, int, types.Integer)
+
+    def __init__(self, func_name):
+        self.func_name = func_name
+
+    def msg(self, ty, val, name=''):
+        if name:
+            name = ' {}'.format(name)
+        return self.msg_template.format(self.func_name, ty, name, val)
+
+    def check_series(self, data):
+        if not isinstance(data, SeriesType):
+            raise TypingError(self.msg('a pandas.series', data))
+
+    def check_integer(self, data, name):
+        if not isinstance(data, (types.Omitted, int, types.Integer)):
+            raise TypingError(self.msg('an integer', data, name=name))
+
+    def check_float(self, data, name):
```

Comment:
all such checks uses predefined list of checked types. This is ambiguous.  Also, types check might be complex (type1 && type2 || type3).
Please do not implement a function for particular check like `check_float`. Please use a list of types as a parameter.
Types or equation will be provided as a parameter.